classes/module/Module.php

```diff
@@ -205,6 +205,9 @@ abstract class ModuleCore implements ModuleInterface
     /** @var array|null used to cache module ids */
     protected static $cachedModuleNames = null;
 
+    /** @var int|null Define the multistore compatibility level of the module */
+    private $multistoreCompatibility;
```

Comment:
You could define an unknown state and get rid of the mixed type
```suggestion
    /** @var int Define the multistore compatibility level of the module */
    private $multistoreCompatibility = self::MULTISTORE_COMPATIBILITY_UNKNOWN;
```